pkg/machines/libvirt-dbus.js

```diff
@@ -1498,6 +1498,10 @@ export function storageVolumeCreate(connectionName, poolName, volName, size, for
                 return call(connectionName, path[0], 'org.libvirt.StoragePool', 'StorageVolCreateXML', [volXmlDesc, 0], TIMEOUT)
                         .then(() => {
                             return storagePoolRefresh(connectionName, path[0]);
+                        }, ex => {
+                            storagePoolRefresh(connectionName, path[0])
+                                    .then(() => storageVolumeDelete(connectionName, poolName, volName));
+                            return Promise.reject(ex);
```

Comment:
Thank you for the fix. 
We should also reuse this fix in CREATE_AND_ATTACH_VOLUME, since that is also place where same bug can happen. Or even better, reuse the whole function storageVolumeCreate in CREATE_AND_ATTACH_VOLUME (sorry that's my fault, I should have edited CREATE_AND_ATTACH_VOLUME to reuse storageVolumeCreate function when I wrote the latter :( ).